Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerImageAttachmentCollectionViewCell.swift

```diff
@@ -5,57 +5,85 @@
 import StreamChat
 import UIKit
 
+/// The view cell that displays the image attachment.
 public typealias ChatMessageComposerImageAttachmentCollectionViewCell =
     _ChatMessageComposerImageAttachmentCollectionViewCell<NoExtraData>
 
+/// The view cell that displays the image attachment.
 open class _ChatMessageComposerImageAttachmentCollectionViewCell<ExtraData: ExtraDataTypes>: _CollectionViewCell,
     UIConfigProvider {
-    // MARK: - Properties
-    
-    class var reuseId: String { String(describing: self) }
-    
+    open class var reuseId: String { String(describing: self) }
+
+    public private(set) lazy var imageAttachmentView = uiConfig
+        .messageComposer
+        .imageAttachmentCellView.init()
+        .withoutAutoresizingMaskConstraints
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+
+        contentView.embed(imageAttachmentView)
+    }
+}
+
+/// A view that displays the image attachment.
+public typealias ChatMessageComposerImageAttachmentView =
+    _ChatMessageComposerImageAttachmentView<NoExtraData>
+
+/// A view that displays the image attachment.
+open class _ChatMessageComposerImageAttachmentView<ExtraData: ExtraDataTypes>: _CollectionViewCell,
+    UIConfigProvider {
+    /// A closure handler that is called when the discard button of the attachment is clicked
     public var discardButtonHandler: (() -> Void)?
-    
-    // MARK: - Subviews
-    
+
+    /// The image view that displays the image of the attachment.
     public private(set) lazy var imageView: UIImageView = UIImageView()
         .withoutAutoresizingMaskConstraints
-    
+
+    /// A button to remove the attachment from the collection of attachments.
     public private(set) lazy var discardButton: UIButton = UIButton()
         .withoutAutoresizingMaskConstraints
-        
-    // MARK: - Lifecycle
-    
+
+    /// The `ChatMessageComposerImageAttachmentView` layout constraints.
+    public struct Layout {
+        public var discardButtonConstraints: [NSLayoutConstraint] = []
+    }
+
+    /// The `ChatMessageComposerImageAttachmentView` layout constraints.
+    public private(set) var layout = Layout()
+
```

Comment:
Do you think we still need this? Complex layouts will be done via `ContainerView`, for simple layouts like this, it's OK to not have it.

Reply:
The reason I did this, is that somehow, and I couldn't find the reason, the only way to override properly the constraints for discard button was to deactivate them. If I didn't, and for example set the discard button to the leading, it would go to the middle of the view 🤷‍♂️